\chapter{Results}\label{chap:results}
\section{Statistical brain shape analysis}

\begin{figure}[H]
	\centering
	\begin{center}
		\makebox[\textwidth]{\includesvg[width=0.95\paperwidth]{symmetryStats/STAGE00DATA/reduce10_gpaReps3/N_REP3/iters10000_nPicks5_perPickSize50_/results.svg}}
	\end{center}
	\caption[Statistical brain shape analysis results]{Asymmetry components significance analysis. Information about each landmark has been used to fill the \ac{3d} average shape of the left hemisphere, displayed medially and laterally. The \acp{mss} are produced by dividing \acp{rss} with the appropriate \acp{dof}. Both the first and the second column refer to information retrieved from the non-permuted data. The last column is derived by applying various thresholds (shown in the colorbar) on the p-value response presented on the third column.}
	\label{fig:stat_analysis}	
\end{figure}

The main results from the 2-way permutation \ac{anova} are summarized in \autoref{fig:stat_analysis} and are also being more formally medically followed by \citet{Vanbiervliet2022}. The individual effect (first row), which corresponds to the average shape variability across individuals, exhibits greater variation in the medial surface, particularly in the middle-anterior and middle-posterior parts of the cingulate gyrus and sulcus ($\subseteq$BA28, processing emotions and behavior regulation), the rostral part of the cuneus ($\subseteq$BA17, processing of visual information), the parahippocampal gyrus ($\subseteq$BA27, memory encoding and retrieval), and the fusiform gyrus ($\subseteq$BA37, recognition of faces). 

\Ac{da} (second row), the focus of this study, relevant to the general aptitude of individuals to exhibit certain asymmetric traits, is found to be highly significant in almost half of the studied surface. In line with the general identified asymmetry patterns presented in \autoref{subsec:general_sym_traits}, it is greatly localized around the peri-sylvian fissure and the temporal lobe (see peri-sylvian asymmetry). Also, it occurs in the medial surface, and the occipital lobe, implying relationship with the yakovlevian torque, although lower significance of the effect is demonstrated on the prefrontal lobe.

\Ac{fa} effect (last row), which, as a reminder, is related to environmentally and developmentally induced variations, has been found generally significant across the cortical surface. This finding can be partly justified to the large combinatorial space from where permutations are collected (see \autoref{subsec:stat_methods}), as well as the overall exhibited plasticity of the human cortex (see \autoref{subsec:plasticity}), as raised by \citet{Vanbiervliet2022}. However, a comparison across regions is possible by inspecting  $F_{FA}$ instead, where, in the caudal part of the middle frontal gyrus ($\subseteq$ BA40, phonological processing and emotional responses), the superior part of the precentral gyrus ($\subseteq$ BA07, space localization), and the caudal part of the superior frontal sulcus and gyrus ($\subseteq$ BA08, planning complex movements) greater effect of \ac{fa} is exhibited.


\section{Covariates control}
In \autoref{fig:covar}, the average covariates explained variance on each segment of \ac{dk} atlas is observed as retrieved from \ac{plsr}. The largest part of the frontal lobe, with greater impact on the inferior frontal, the inferior parietal gyrus and the parahippocampal gyrus appear to be more correlated with the collected metadata, shown in \autoref{tab:covariates}. The results point to less observed explained variation around the area of the sylvian fissure and the temporal lobe, and also raise a degree of uncertainty on the significant response observed during the statistical shape asymmetry analysis, regarding the medial surface and the inferior part of the frontal lobe.
\begin{figure}[H]
\includesvg[width=0.9\textwidth]{asymmetry/hierarchicalClusteringDemo/STAGE00DATA/asymmetry_reduction1/explainedDKCovariates.svg}
\caption[Explained variance from covariates]{Explained variance from the covariates on each \ac{dk} atlas segment, as retrieved from \ac{plsr}, mapped on the average left hemisphere medial and lateral side.}
\label{fig:covar}
\end{figure}
\section{Partitioning and \ac{pca}}
In \autoref{fig:partitioning}, the partitioning produced from the application of \ac{hsc} is displayed. On the first level, although the cross-section accurately follows the sylvian fissure on the lateral part and partitions the frontal lobe from the rest of the hemisphere, on the medial surface it appears to split the precuneus in half ($\subseteq$ BA07). On the second level, the occipital lobe is separated from the temporal lobe, while the central gyrus appears to be dissected from the frontal lobe, while inspected the lateral surface. On the medial surface, the paracentral gyrus ($\subseteq$ BA04)is approximately separated from the superior-frontal, whereas another cross section appears to share its boundaries with the temporal pole ($\subseteq$ BA38). In general, the unsupervised clustering follows the functional partitioning, validating the close relationship  between function and morphology of each cortical region. The calculated \ac{nmi} score for the partitioning, compared to the \ac{dk} atlas is displayed in \autoref{tab:nmi} for each level. Although the finer the partitioning, the further away from the theoretical maximum value, quantitatively the two clusterings highly agree.

\begin{figure}[H]
	\begin{adjustwidth}{-4cm}{-4cm}
	\centering
	\subfloat[]{
		\includesvg[width=0.6\linewidth]{asymmetry/hierarchicalClusteringDemo/STAGE00DATA/asymmetry_reduction1/levels4/segmentation.svg}
	}\quad
	\subfloat[]{
		\includesvg[width=0.4\linewidth]{asymmetry/hierarchicalClusteringDemo/STAGE00DATA/asymmetry_reduction1/levels4/segmentation_circular.svg}
	}

\end{adjustwidth}
\caption[4-level brain shape partitioning based on asymmetry]{4-level brain shape partitioning based on cortical asymmetry, using \ac{hsc}. Shown in 2 different versions, on a level representation (top), and as a polar dendrogram plot, annotated with white color against black background (bottom). Those representations are used across the coarse-to-fine analysis in this study.}
\label{fig:partitioning}
\end{figure}

\begin{table}[H]
	\centering
	\begin{tabular}{l|ccc}
		& NMI & $\NMI_{max}$ & ratio\\
		\hline
		 Lvl 1 & 0.37 & 0.48 & 0.78\\
		 Lvl 2 & 0.49 & 0.66 & 0.74\\
		 Lvl 3 & 0.55 & 0.79 & 0.71\\
		 Lvl 4 & 0.62 & 0.90 & 0.69\\
	\end{tabular}
\caption [\ac{nmi} scores across \ac{hsc} partitioning levels]{\Ac{nmi} scores across \ac{hsc} partitioning levels, comparing \ac{dk} atlas with computed partitioning levels. $\NMI_{max}$ is an approximate maximal value, given the different number of partitions in each clustering, and ratio is the scaled \ac{nmi} using that value.}
\label{tab:nmi}
\end{table}
The required number of \ac{pca} features per partition given the constraints, as computed by assessing the discovery dataset, is displayed in \autoref{fig:pcs_num}. A significant dimensionality reduction was achieved, given that the original total number of landmark coordinates per individual was 89367.
\begin{figure}[H]
\includesvg[width=0.9\textwidth]{asymmetry/hierarchicalClusteringDemo/STAGE00DATA/asymmetry_reduction1/levels4/explainedVarianceSIngleThreshold.svg}
\caption[Number of PCs per HSC cortical surface partition]{Number of \acp{pc} for each \ac{hsc} cortical surface partition, required to explain 80\% of its variance, relatively to the discovery dataset. For the first partition, the upper limit of 500 components is reached and only 74\% of its variance is explained.}
\label{fig:pcs_num}
\end{figure}
\section{\ac{gwas}}

The resulting \ac{mvgwas} results on the entire hemisphere exhibit distinct peaks, that slightly differ between discovery and replication dataset (\autoref{fig:entire_gwas}). 22 lead \acp{snp} with p-value less than 5e-8, versus 21 reported in literature \cite{Sha2021}, are retrieved from the meta-analysis union, using the methods defined earlier (\autoref{subsec:func_mat_methods}), six of which have also been exactly identified by \citet{Sha2021} (\autoref{tab:leadsnps}). Additionally, the number of significant variants after Bonferroni correction is reported in \autoref{fig:part_bonferroni}. A different chromosome-based response is observed when inspecting the second level of partitioning, namely partitions 4-7, so a further analysis has been deemed necessary on each of those partitions separately. Furthermore, the partition 6, which corresponds to the frontal lobe cortical asymmetry, appears to be quite less genetically determined and agrees with the findings from the statistical analysis, where \ac{da} did not appear to be in general significant in that region, whereas \ac{fa} did (\autoref{fig:stat_analysis}). An additional observation is that partitions do not necessarily share the subset of the genetic signature of their parents, for example partition 8 exhibit a significant peak at chromosome 21, while that chromosome has not been identified elsewhere to exhibit significant genetic effect on cortical asymmetry . \citet{Sha2021} reported the most significant \ac{snp}, rs41298373 (P=5e-38), to be located in chromosome 10, although in this study, chromosome 10 does not present any significant signal whatsoever. Chromosome 15 exhibits the highest peak, rs2033939 (P=2.09e-50), but no peaks for that chromosome were identified by \citet{Sha2021}. In general, several differences with the primary results reported by \citet{Sha2021} are observable in the current study, even from a qualitative viewpoint. There is also generally more confidence reported over common identified lead \acp{snp}, characterized by lower p-values in the present study than the ones reported in literature \cite{Sha2021}. A more refined, quantitative comparison is also performed by applying Spearman correlation, with the way mentioned in \autoref{subsec:gencorr} \cite{Naqvi2021}.
\begin{adjustwidth}{-3 cm}{-2.5 cm}\centering\begin{threeparttable}[!htb]
		
		\scriptsize
		\begin{tabular}{ccc||ccc||ccc||ccc}\toprule
			\multicolumn{3}{c}{\textbf{DISCOVERY}} &\multicolumn{3}{c}{\textbf{REPLICATION}} &\multicolumn{3}{c}{\textbf{UNION}} &\multicolumn{3}{c}{\textbf{SHA ET AL.}} \\\midrule
			\textbf{CHR} &\textbf{SNP} &\textbf{P} &\textbf{CHR} &\textbf{SNP} &\textbf{P} &\textbf{CHR} &\textbf{SNP} &\textbf{P} &\textbf{CHR} &\textbf{SNP} &\textbf{P} \\
			15 &rs1080066 &4.10e-28 &15 &rs4924345 &1.41e-27 &15 &rs2033939 &2.09e-50 &10 &\textbf{rs41298373} &5.00e-38 \\
			17 &rs10775404 &5.12e-27 &17 &rs9910683 &2.42e-20 &17 &17:44243312\_C\_G &3.91e-44 &17 &rs35853889 &1.00e-20 \\
			14 &rs160459 &2.57e-17 &9 &rs7023465 &3.74e-18 &14 &\textbf{rs160459} &3.58e-28 &17 &rs80103986c &5.00e-16 \\
			1 &rs6658111 &3.05e-15 &14 &rs74826997 &3.02e-14 &9 &rs7023465 &1.40e-25 &9 &rs911934 &2.00e-15 \\
			16 &rs111398992 &1.17e-12 &14 &rs160459 &6.57e-13 &1 &\textbf{rs6658111} &4.31e-25 &17 &rs55938136 &5.00e-15 \\
			2 &rs7575465 &1.57e-12 &1 &rs6658111 &8.08e-12 &14 &rs74826997 &3.36e-22 &16 &\textbf{rs111398992} &6.00e-15 \\
			16 &rs9939914 &7.01e-12 &10 &rs41298373 &2.10e-11 &16 &\textbf{rs72813426} &3.78e-21 &17 &rs568039055 &8.00e-15 \\
			10 &rs41298373 &3.66e-11 &16 &rs35769864 &1.40e-10 &10 &\textbf{rs41298373} &9.26e-21 &16 &\textbf{rs72813426} &2.00e-14 \\
			17 &rs62063672 &1.40e-10 &12 &rs10783306 &7.41e-10 &16 &\textbf{rs111398992} &1.66e-20 &14 &\textbf{rs160459} &5.00e-12 \\
			3 &rs35124509 &2.71e-10 &2 &rs7575465 &7.75e-10 &2 &rs7575465 &1.80e-20 &12 &rs10783306 &1.00e-11 \\
			14 &rs76341705 &3.81e-10 &16 &rs111398992 &9.24e-10 &3 &rs35124509 &1.26e-15 &2 &rs12617392 &4.00e-11 \\
			9 &rs911934 &4.71e-10 & & & &16 &rs11640500 &2.84e-15 &1 &\textbf{rs6658111} &1.00e-10 \\
			3 &rs2399037 &4.21e-9 & & & &12 &rs1057548 &1.71e-13 &2 &rs62130503 &1.00e-10 \\
			7 &rs117766132 &6.39e-9 & & & &16 &rs12711472 &4.61e-12 &2 &rs368536282 &1.00e-10 \\
			\cellcolor[HTML]{C8C8C8}2 &\cellcolor[HTML]{C8C8C8}rs2691112 &\cellcolor[HTML]{C8C8C8}1.29e-8 & & & &2 &rs7601767 &8.67e-11 &6 &rs7781 &2.00e-10 \\
			\cellcolor[HTML]{C8C8C8}2 &\cellcolor[HTML]{C8C8C8}rs7601767 &\cellcolor[HTML]{C8C8C8}1.88e-8 & & & &7 &rs151218739 &1.76e-10 &14 &rs201816193 &4.00e-10 \\
			\cellcolor[HTML]{C8C8C8}8 &\cellcolor[HTML]{C8C8C8}rs7000137 &\cellcolor[HTML]{C8C8C8}2.60e-8 & & & &8 &rs10106820 &1.80e-10 &19 &rs11672092 &6.00e-10 \\
			& & & & & &2 &rs13021393 &6.80e-10 &21 &\textbf{rs7283026} &8.00e-10 \\
			& & & & & &21 &\textbf{rs7283026} &3.01e-9 &3 &rs2279829 &1.00e-9 \\
			& & & & & &8 &rs442103 &3.43e-9 &5 &rs869219775 &3.00e-9 \\
			& & & & & &15 &rs8026370 &4.49e-9 &20 &rs6135555 &7.00e-9 \\
			& & & & & &21 &rs4819075 &5.01e-9 &\cellcolor[HTML]{C8C8C8}X &\cellcolor[HTML]{C8C8C8}rs12400461 &\cellcolor[HTML]{C8C8C8}1.00e-8 \\
			& & & & & &\cellcolor[HTML]{C8C8C8}2 &\cellcolor[HTML]{C8C8C8}rs2287283 &\cellcolor[HTML]{C8C8C8}1.38e-8 &\cellcolor[HTML]{C8C8C8}6 &\cellcolor[HTML]{C8C8C8}rs9385385 &\cellcolor[HTML]{C8C8C8}1.00e-8 \\
			& & & & & &\cellcolor[HTML]{C8C8C8}2 &\cellcolor[HTML]{C8C8C8}rs7609463 &\cellcolor[HTML]{C8C8C8}1.38e-8 &\cellcolor[HTML]{C8C8C8}17 &\cellcolor[HTML]{C8C8C8}rs35908989 &\cellcolor[HTML]{C8C8C8}1.00e-8 \\
			& & & & & &\cellcolor[HTML]{C8C8C8}17 &\cellcolor[HTML]{C8C8C8}rs930119 &\cellcolor[HTML]{C8C8C8}3.02e-8 &\cellcolor[HTML]{C8C8C8}4 &\cellcolor[HTML]{C8C8C8}rs9307052 &\cellcolor[HTML]{C8C8C8}2.00e-8 \\
			& & & & & &\cellcolor[HTML]{C8C8C8}11 &\cellcolor[HTML]{C8C8C8}rs2251662 &\cellcolor[HTML]{C8C8C8}3.41e-8 &\cellcolor[HTML]{C8C8C8}22 &\cellcolor[HTML]{C8C8C8}rs9615351 &\cellcolor[HTML]{C8C8C8}3.00e-8 \\
			& & & & & & & & &\cellcolor[HTML]{C8C8C8}7 &\cellcolor[HTML]{C8C8C8}rs6947352 &\cellcolor[HTML]{C8C8C8}4.00e-8 \\
			\bottomrule
		\end{tabular}
		\caption[Identified lead \acp{snp} for the entire hemisphere]{Identified lead \acp{snp} for the entire hemisphere across the different datasets, along with a comparison with \citet{Sha2021}. In gray, \acp{snp} with significance less than 1e-8 but greater than 5e-8 are reported. In bold are displayed the variants identified both in the present study and in the work of \citet{Sha2021}.}
		\label{tab:leadsnps}
\end{threeparttable}\end{adjustwidth}
\begin{figure}[H]
	\centering
	\subfloat{
		\includesvg[width=1\textwidth]{asymmetry/genomeDemo/STAGE00DATA/mean_imputed/not_subsampled/gwas_discovery.svg}
	}
	\par\medskip
	\centering
	\subfloat{
		\includesvg[width=1\textwidth]{asymmetry/genomeDemo/BATCH2_2021_DATA/mean_imputed/not_subsampled/gwas_replication.svg}	
	}
	\par\medskip
	\centering
	\subfloat{
		\includesvg[width=1\textwidth]{asymmetry/meta_analysis/joinedDatasets/mean_imputed/not_subsampled/gwas_joined.svg}	
	}
	\caption[mvGWAS of the entire hemisphere shape asymmetry]{\Ac{mvgwas} of the entire hemisphere shape asymmetry computed on the discovery (top) and the replication (middle) dataset, along with the meta-analysis union based on Stouffer's method (bottom).}
	\label{fig:entire_gwas}
\end{figure}




\begin{figure}[H]
	\begin{adjustwidth}{-4cm}{-3cm}
	\centering
	\subfloat{
		\includesvg[width=0.85\linewidth]{asymmetry/visualizeCCAOnPheno/joinedDatasets/mean_imputed/not_subsampled/WithBC/CcaCircular.svg}
	}
	\subfloat{
		\includesvg[width=0.15\linewidth]{asymmetry/visualizeCCAOnPheno/joinedDatasets/mean_imputed/not_subsampled/WithBC/legend.svg}
	}
	\end{adjustwidth}	
	\caption[Number of significant SNPs after Bonferroni correction along partitions]{Number of significant \acp{snp} per chromosome after Bonferroni correction across partitions. The number per chromosome is shown inside boxes of a certain color.}
	\label{fig:part_bonferroni}
		
\end{figure}
\section{\ac{ldsr}}
\section{\ac{ldsc}}
\section{\ac{ldsc}-SEG}
\section{Developmental analysis}
\section{Functional association}
\section{Evolutionary studies}

